### DISCUSSION TITLE
Access UI in Phaser Version of Engine

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#144] [Help / Questions] [answered]
How would I access the `ui` element of the scene when using the phaser version of the engine? I tried the `.third.ui`, but that was undefined.
I'm using the `enable3d.phaserExtension.0.22.0-dev.1.min.js` version of the library.
Thanks!

**Comment by yandeu** [comment, marked as answer]
`ui` is not defined by default. Take a look at the examples.

Also, I'm not sure if the Phaser Extension exposes `FLAT`.

**Reply by N8python** [reply]
Yeah I tried doing the examples and I don't think its compatible with the phaser version - I just used an html/css overlay. Thanks anyway!

**Reply by yandeu** [reply]
It looks like the Phaser Extension has the `FLAT` module.

See:
https://github.com/enable3d/enable3d/blob/master/packages/phaserExtension/src/index.ts

Maybe try dev.2:
https://github.com/enable3d/enable3d/tree/master/bundles

I will test it later this day.

**Reply by N8python** [reply]
It's fine - I found a working solution. Don't trouble yourself. I think I was doing it wrong anyway.

**Reply by yandeu** [reply]
I also found a way:

```ts
const { enable3d, Scene3D, Canvas, FLAT, THREE } = ENABLE3D

class MainScene extends Scene3D {
  init() {
    this.accessThirdDimension()
  }

  create() {
    this.third.warpSpeed()
    this.third.physics.add.box()

    const width = window.innerWidth
    const height = window.innerHeight

    // 2d camera
    this.third.renderer.autoClear = false // To allow render overlay on top of the 3d camera
    this.ui = {
      camera: this.third.cameras.orthographicCamera({ left: 0, right: width, bottom: 0, top: height }),
      scene: new THREE.Scene()
    }

    const texture = new FLAT.TextTexture('Some Text')
    const sprite2d = new FLAT.TextSprite(texture)
    this.ui.scene.add(sprite2d)
    sprite2d.setPosition(sprite2d.width / 2 + 40, height - sprite2d.height / 2 - 40)

    this.third.preRender = () => {
      // needed for the 2d camera
      this.third.renderer.clear()
    }

    this.third.postRender = () => {
      // needed for the 2d camera
      if (this.ui) {
        this.third.renderer.clearDepth()
        this.third.renderer.render(this.ui.scene, this.ui.camera)
        FLAT.render(this.ui.camera)
      }
    }
  }
}
```
